### PR TITLE
filter activities only for synced or owned repositories

### PR DESCRIPTION
**Pull request by Chilledheart**

### CODE REVIEW

File: src/ui/activities-tab.cpp [hunk 1]
```diff
@@ -79,7 +82,21 @@ void ActivitiesTab::refreshEvents(const std::vector<SeafEvent>& events,
...
+    // TODO use a option to control this filter
+    std::vector<SeafEvent> filtered_events;
+    LocalRepo r;
+    ServerRepo repo;
+    for (size_t i = 0; i < events.size(); ++i) {
+        const QString &repo_id = events[i].repo_id;
+        repo = RepoService::instance()->getRepo(repo_id);
+        if ((repo.isValid() && repo.owner == seafApplet->accountManager()->currentAccount().username) ||
+            // TODO use RepoService::instace()->getLocalRepo to avoid RPC call
+            (seafApplet->rpcClient()->getLocalRepo(repo_id, &r) == 0 && r.isValid())) {
```
Comment (lins05) [inline src/ui/activities-tab.cpp]: You can use `SeafileRpcClient::hasLocalRepo(repo_id)` here.

Comment (Chilledheart) [inline src/ui/activities-tab.cpp]: done

**Comment by lins05** [comment]
The code looks good to me. But what if all the events are filtered out, the events tab would show nothing, right? I think we can request more events from the server until we have at enough items.

**Comment by Chilledheart** [comment]
@lins05 Well, it is more complicate to support with `load more events` at client side but not impossible. One more question, why those events are not filtered by the server side while we just need the filtered ones only? It seems more reasonable to filter events at the server side anyway.

**Comment by Chilledheart** [comment]
It seems I get the task misunderstood, just closing it.

@lins05 just let me know if we are supported to add `load more events`. It looks like great.

**Pull request closed without merging**